app/models/host/managed.rb

```diff
@@ -314,12 +314,20 @@ def handle_realm
     end
   end
 
+  def disk_layout_template
+    if disk.present?
+      { name: 'Custom disk layout', template: disk }
+    elsif ptable.present?
+      { name: ptable.name, template: ptable.layout }
+    end
```

Comment:
Not very significant but maybe cache this hash to prevent multiple object allocations on each call to `diskLayout` etc?

Reply:
I would prefer not to cache until proved worth it.

Reply:
Now that I read the code again I (think I) understand why I added this comment a year ago - `diskLayout` calls this function 3 times, so each call this function is executed again, instead of saving the result in a variable in that function. That said, I don't think this is crucial and won't block on it.